scss/forms/_floating-labels.scss

```diff
@@ -0,0 +1,65 @@
+// stylelint-disable selector-no-vendor-prefix
+
+.form-floating {
+  position: relative;
+
+  > label,
+  > .form-control,
+  > .form-select {
+    height: $form-floating-height;
+    padding: $form-floating-padding-y $form-floating-padding-x;
+  }
+
+  > label {
+    position: absolute;
+    top: 0;
+    left: 0;
+    display: block;
+    width: 100%;
+    margin-bottom: 0; // Override default `<label>` margin
+    pointer-events: none;
+    cursor: text; // Match the input under the label
+    border: $input-border-width solid transparent;
```

Comment:
Why do we have this transparent border?

Reply:
Added a code comment—it affects the box model and shifts the text.